.github/workflows/build.yml

```diff
@@ -405,13 +483,29 @@ jobs:
     steps:
       - uses: actions/checkout@v2
         if: ${{ matrix.projects != 'dummy' }}
+
+      - name: Cache for generated files
+        uses: actions/cache@v2
+        with:
+          path: generated_files.tar.gz
+          key: ${{ runner.os }}-${{ hashFiles('libs/api/domains/cms/src/lib/generated/contentfulTypes.d.ts', 'apps/**/models/**/*.tsx?', 'libs/**/models/**/*.tsx?', 'apps/**/dto/**/*.tsx?', 'libs/**/dto/**/*.tsx?', 'apps/**/enums/**/*.tsx?', 'libs/**/enums/**/*.tsx?', 'apps/**/queries/**/*.tsx?', 'libs/**/queries/**/*.tsx?', 'apps/air-discount-scheme/web/screens/**/*.tsx', 'apps/air-discount-scheme/web/i18n/withLocale.tsx', 'apps/air-discount-scheme/web/components/AppLayout/AppLayout.tsx', 'apps/air-discount-scheme/web/components/Header/Header.tsx', 'apps/gjafakort/api/src/**/typeDefs.ts') }}-generated-files
```

Comment:
Would like to see a variable for this, which can be reused for all the keys. Something like:
```yaml
cache_key: &cache_key "hashFiles('libs/api/domains/cms/src/lib/generated/contentfulTypes.d.ts', 'apps/**/models/**/*.tsx?', 'libs/**/models/**/*.tsx?', 'apps/**/dto/**/*.tsx?', 'libs/**/dto/**/*.tsx?', 'apps/**/enums/**/*.tsx?', 'libs/**/enums/**/*.tsx?', 'apps/**/queries/**/*.tsx?', 'libs/**/queries/**/*.tsx?', 'apps/air-discount-scheme/web/screens/**/*.tsx', 'apps/air-discount-scheme/web/i18n/withLocale.tsx', 'apps/air-discount-scheme/web/components/AppLayout/AppLayout.tsx', 'apps/air-discount-scheme/web/components/Header/Header.tsx', 'apps/gjafakort/api/src/**/typeDefs.ts')"
```

then it can be referenced with:
```yaml
key: ${{ runner.os }}-${{ *cache_key }}-generated-files
```
(something on these lines)

Reply:
Will developers have to edit this variable when they create a new project?

Reply:
That's a good question and this is the weakest part of the script.

I added the folders (the one that I saw) that trigger the scripts to generate new files (autoSchemeFile/openApi etc). Most of the projects have common folders where it happens e.g. `queries/*`, `models/*`, etc but it's not always the case.

If a new project is created and follows the most common way to create queries/models etc, it will be okay. However, if some files at some specific paths must trigger newly generated files and it's not part of the wildcard it will fail to generate the new schemas.

An alternative would be to never cache theses files on the build steps with the hash key and generate them all the time, but it will increase the build time of ~30sec * number of steps needed

cc/ @eirikurn @sindrig

Reply:
surprise surprise I keep forgetting about yaml variables. we should look into creating a parameter node in the build's root for this.

as for if developers will have to edit this variable. like @jeremybarbet says, if the most common way to create queries/models is used, there will be no problem. and if a developer uses some obscure way, the build will fail and he/she will be faced with the decision to either update this variable (and the .gitignore and removal script probably) or simply use the common way.

Reply:
Unfortunately, GitHub actions don't support anchors :scream_cat: 
https://github.com/island-is/island.is/issues/1117